### ISSUE TITLE
AlignerNet

### ISSUE THREAD

**Issue by Tera2Space** [p0p4k/pflowtts_pytorch#22]
So i noticed commented aligner in pflow code, you commented it because it didn't work or it didn't improve quality much?

**Comment by Tera2Space** [comment]
Yep, i tried but loss stuck at around 6, i think it's because we give to it full output of the encoder(mu) instead of giving only output of text_base_encoder (like in naturalspeech), can it be the reason ?

**Comment by p0p4k** [comment]
I am not sure if I understand your question.. if you want to edit and paste the code here, i might understand better.

**Comment by p0p4k** [comment]
If MAS can align it with mu_x, the alignernet should be able to do it too. It will take longer to converge though.

**Comment by Tera2Space** [comment]
and last question, for Alignernet to work I only need to uncomment
```python

**Comment by Tera2Space** [comment]
And last question, for AlignerNet to work I only need to uncomment:
```python 
        # self.aligner = Aligner(
        #     dim_in=encoder.encoder_params.n_feats,
        #     dim_hidden=encoder.encoder_params.n_feats,
        #     attn_channels=encoder.encoder_params.n_feats,
        #     )
        
        # self.aligner_loss = ForwardSumLoss()
        # self.bin_loss = BinLoss()
        # self.aligner_bin_loss_weight = 0.0
```
and
```python
        # aln_hard, aln_soft, aln_log, aln_mask = self.aligner(
        #     mu_x.transpose(1,2), x_mask, y, y_mask
        #     )
        # attn = aln_mask.transpose(1,2).unsqueeze(1)
        # align_loss = self.aligner_loss(aln_log, x_lengths, y_lengths)
        # if self.aligner_bin_loss_weight > 0.:
        #     align_bin_loss = self.bin_loss(aln_mask, aln_log, x_lengths) * self.aligner_bin_loss_weight
        #     align_loss = align_loss + align_bin_loss
        # dur_loss = F.l1_loss(logw, attn.sum(2))
        # dur_loss = dur_loss + align_loss
```
and comment MAS usage, right?

**Comment by p0p4k** [comment]
Yes correct.

**Comment by Tera2Space** [comment]
Great, thanks a lot for the answers!

**Comment by UESTC-Med424-JYX** [comment]
Hello, how is the training speed of your AlignerNet? After I added this module, the training speed became very slow. After debugging, I found that the problem occurred in:
`alignment_mask = maximum_path(alignment_soft, attn_mask)`
What is the impact on the results if this loss is included?
